### ISSUE TITLE
Replace CTRL-click for free-hand transforms with re-click & icon change

### ISSUE THREAD

**Issue by sofian** [mapmapteam/mapmap#372]
Currently MapMap supports free-hand transforms using CTRL-click (#297):
https://www.youtube.com/watch?v=EORy4UbQeWQ

However this is not very intuitive. Instead we want the user to be able to switch between 3 modes:
- translation (ie. default)
- rotate
- scale

To change mode the user would simply re-click on the same shape. This is similar to the functionality in Inkscape:
https://www.youtube.com/watch?v=J_mN2VY2_ks

To indicate that we are in a certain mode, we would simply change the "look" of the corner buttons. For example:

Translate (default, current):
![cornericon](https://user-images.githubusercontent.com/791244/36908953-c477f6a6-1e0a-11e8-8793-e73f7d74ccc3.png)

Rotate:
![cornericon_scale](https://user-images.githubusercontent.com/791244/36908951-c450e4b2-1e0a-11e8-8256-eaf45a08c093.png)

Scale:
![cornericon_rotate](https://user-images.githubusercontent.com/791244/36908952-c4649c1e-1e0a-11e8-8a37-afb8a7801247.png)

@baydam I can ask someone to help us with the icons if needed.

**Comment by loshermanos** [comment]
Hello i can work on it,

The scale button looks like a rotate button.

________________________________
De : Sofian Audry <notifications@github.com>
Envoyé : vendredi 2 mars 2018 16:16
À : mapmapteam/mapmap
Cc : Subscribed
Objet : [mapmapteam/mapmap] Replace CTRL-click for free-hand transforms with re-click & icon change (#372)


Currently MapMap supports free-hand transforms using CTRL-click (#297<https://github.com/mapmapteam/mapmap/issues/297>):
https://www.youtube.com/watch?v=EORy4UbQeWQ

However this is not very intuitive. Instead we want the user to be able to switch between 3 modes:

  *   translation (ie. default)
  *   rotate
  *   scale

To change mode the user would simply re-click on the same shape. This is similar to the functionality in Inkscape:
https://www.youtube.com/watch?v=J_mN2VY2_ks

[https://www.bing.com/th?id=OVP.CyZBjek-8vEtjNkiiRr87QEsCo&pid=Api]<https://www.youtube.com/watch?v=J_mN2VY2_ks>

Rotating and Skewing an Object in Inkscape<https://www.youtube.com/watch?v=J_mN2VY2_ks>
www.youtube.com
Learn about Rotating and Skewing an Object in Inkscape



To indicate that we are in a certain mode, we would simply change the "look" of the corner buttons. For example:

Translate (default, current):
[cornericon]<https://user-images.githubusercontent.com/791244/36908953-c477f6a6-1e0a-11e8-8793-e73f7d74ccc3.png>

[https://user-images.githubusercontent.com/791244/36908953-c477f6a6-1e0a-11e8-8793-e73f7d74ccc3.png]

Rotate:
[cornericon_scale]<https://user-images.githubusercontent.com/791244/36908951-c450e4b2-1e0a-11e8-8256-eaf45a08c093.png>

[https://user-images.githubusercontent.com/791244/36908951-c450e4b2-1e0a-11e8-8256-eaf45a08c093.png]

Scale:
[cornericon_rotate]<https://user-images.githubusercontent.com/791244/36908952-c4649c1e-1e0a-11e8-8a37-afb8a7801247.png>

[https://user-images.githubusercontent.com/791244/36908952-c4649c1e-1e0a-11e8-8a37-afb8a7801247.png]

@baydam<https://github.com/baydam> I can ask someone to help us with the icons if needed.

**Comment by sofian** [comment]
Yeah these were just quickly generated they are by no means meant to be permanent. Would be great if you could make something @loshermanos !

Also: I believe we should find a way for these icons (if possible) to indicate the center of the icon. Our current corner icons do a good job at that cause the center of the "X" marks the center of the circle. This is helpful when you want to align the corner with something on the scene. Would be important to have this for the rotate and scale corners as well.

Also, while we're at it, in order to keep things uniform, I would suggest you also design an image for the current "translate" corners (have a look in the interface, the button is actually not a cross (+) but a "X"). And please make a SVG **and** a hi-res PNG versions (with transparency).

**Comment by sofian** [comment]
Also would be neat if we could have icons for the menu items. We could also have (from the menu items only) a 4th "mode" which would do "scale & rotate" at the same time (which can be really useful). We'd need an icon for that as well.

**Comment by loshermanos** [comment]
Ok i ll work on it sunday.

yesterday with Bay we ve working quickly on 3 buttons : the source and destination layout + both




________________________________
De : Sofian Audry <notifications@github.com>
Envoyé : vendredi 2 mars 2018 16:46
À : mapmapteam/mapmap
Cc : loshermanos; Mention
Objet : Re: [mapmapteam/mapmap] Replace CTRL-click for free-hand transforms with re-click & icon change (#372)


Yeah these were just quickly generated they are by no means meant to be permanent. Would be great if you could make something @loshermanos<https://github.com/loshermanos> !

Also: I believe we should find a way for these icons (if possible) to indicate the center of the icon. Our current corner icons do a good job at that cause the center of the "X" marks the center of the circle. This is helpful when you want to align the corner with something on the scene. Would be important to have this for the rotate and scale corners as well.

Also, while we're at it, in order to keep things uniform, I would suggest you also design an image for the current "translate" corners (have a look in the interface, the button is actually not a cross (+) but a "X"). And please make a SVG and a hi-res PNG versions (with transparency).

**Comment by aalex** [comment]
@loshermanos Can you send us the SVG files? Thanks!!

**Comment by loshermanos** [comment]
Here is a test for the scale - rotation translate button
tell me wath you thing
Cheers
M.

________________________________
De : Sofian Audry <notifications@github.com>
Envoyé : vendredi 2 mars 2018 16:16
À : mapmapteam/mapmap
Cc : Subscribed
Objet : [mapmapteam/mapmap] Replace CTRL-click for free-hand transforms with re-click & icon change (#372)


Currently MapMap supports free-hand transforms using CTRL-click (#297<https://github.com/mapmapteam/mapmap/issues/297>):
https://www.youtube.com/watch?v=EORy4UbQeWQ

However this is not very intuitive. Instead we want the user to be able to switch between 3 modes:

  *   translation (ie. default)
  *   rotate
  *   scale

To change mode the user would simply re-click on the same shape. This is similar to the functionality in Inkscape:
https://www.youtube.com/watch?v=J_mN2VY2_ks

To indicate that we are in a certain mode, we would simply change the "look" of the corner buttons. For example:

Translate (default, current):
[cornericon]<https://user-images.githubusercontent.com/791244/36908953-c477f6a6-1e0a-11e8-8793-e73f7d74ccc3.png>

Rotate:
[cornericon_scale]<https://user-images.githubusercontent.com/791244/36908951-c450e4b2-1e0a-11e8-8256-eaf45a08c093.png>

Scale:
[cornericon_rotate]<https://user-images.githubusercontent.com/791244/36908952-c4649c1e-1e0a-11e8-8a37-afb8a7801247.png>

@baydam<https://github.com/baydam> I can ask someone to help us with the icons if needed.

**Comment by sofian** [comment]
Hey @loshermanos Where did you put the logos? I cannot find them.

**Comment by baydam** [comment]
I pushed a new branch where I replace the Ctrl+click by three click state for translate, scale and rotate
https://github.com/mapmapteam/mapmap/tree/feature/transform-shape-gui
I actually need the icons for complete the user integration
